.github/workflows/static_analysis.yml

```diff
@@ -40,13 +33,32 @@ jobs:
           submodules: false
       - name: Setup and run tests
         shell: bash
+        env:
+          TASK: lint
         run: |
-          export TASK="${{ matrix.task }}"
-          export BUILD_DIRECTORY="$GITHUB_WORKSPACE"
           export CONDA=${HOME}/miniforge
           export PATH=${CONDA}/bin:$HOME/.local/bin:${PATH}
           $GITHUB_WORKSPACE/.ci/setup.sh || exit 1
-          $GITHUB_WORKSPACE/.ci/test.sh || exit 1
+          $GITHUB_WORKSPACE/.ci/lint-all.sh || exit 1
+  python-check-docs:
+    name: python-check-docs
```

Comment:
Our docs are not only Python docs... For example, they include C API and general info about LightGBM. Maybe it's better to omit `python` prefix?